code/modules/robotics/robot/upgrade/sechud_scanner.dm

```diff
@@ -0,0 +1,44 @@
+/obj/item/roboupgrade/sechudgoggles
+	name = "cyborg Security HUD upgrade"
+	desc = "An advanced sensor array that allows a cyborg to quickly determine the physical condition of organic life."
+	icon_state = "up-sechud"
+	drainrate = 5
+
+	var/client/assigned = null
+
+	proc/updateIcons()
+		// I wouldve liked to avoid this but i dont want to put this inside the mobs life proc as that would be more code.
+		// TODO: signals/components!
+		while (src.assigned)
```

Comment:
I would subscribe to item processing and use that instead, this is not good and will never stop even if it's outside a bot

Reply:
Got a small example or somewhere i could find where something like this is used? Both sechud and prodocs seem to just work on the process loop.